### DISCUSSION TITLE
Why fork from Urho3D?

### DISCUSSION THREAD

**Discussion by klaussilveira** [u3d-community/U3D#3] [General]
U3D is standing on the shoulders of giants: it is a fork from the last known stable version of [Urho3D](https://github.com/urho3d/Urho3D). The original engine has been going through a lot of changes, most of them with the potential of disrupting the workflow of current and future projects using it. U3D aims to stabilize the development of the engine, focusing on maintainability and continuing the legacy of Urho3D. 

On the opposite direction, there is [rbfx](https://github.com/rbfx/rbfx). Those guys are pushing the engine into the future, with an amazing renderer, integrated editor, glTF standard support, and much more. U3D aims to be more conservative, rbfx is bleeding edge. 

A big thank you to @eugeneko, @rokups and @gleblebedev for all their hardwork and encouraging other members of the Urho community to move forward.

**Comment by gleblebedev** [comment]
I can't take credit for rbfx, it was mostly the other two :-)

**Comment by 1vanK** [comment]
Can you compete with fork https://gitlab.com/luckeyproductions/dry ? This is also a victim of my dictatorship. I think cooperation will double your progress.

**Comment by rokups** [comment]
> Can you compete with fork https://gitlab.com/luckeyproductions/dry ? This is also a victim of my dictatorship. I think cooperation will double your progress.

You had your hostile takeover and you won. Learn to enjoy your victory (however scummy and undeserved) and move on with your russian community. Forks are there precisely because nobody wants to do anything with you.

**Comment by advnau** [comment]
> U3D aims to stabilize the development of the engine, focusing on maintainability and continuing the legacy of Urho3D.
> U3D aims to be more conservative, rbfx is bleeding edge.
> 

I am still curious as to what the end goal is for this project. These lines would make me think the goal is to focus only on UI/UX, dependency updating, and bug fixes. I do not want to understate how appreciated that will be for Urho users. However, if that is the case, we will still be in a state of stagnation.

Is there a roadmap or plan for any future features, new additions, or engine expansions?

**Reply by eugeneko** [reply]
> However, if that is the case, we will still be in a state of stagnation.

I don't want to sound like advertiser, but if you want an Urho3D which is not stagnating, you can as well fork rbfx and try to tweak it.
Urho3D itself is obsolete and outdated, and it will take *a lot* of effort to change that.

If you are okay with that, you can as well stick with "Urho3D on life support" aka U3D.
If you want something up-to-date, you better fork from a project which is already halfway there.
Forking from obsolete project and wasting time re-doing the upgrades... is suboptimal, to say the least.

**Reply by klaussilveira** [reply]
@eugeneko you are right, though. https://github.com/rbfx/rbfx is definitely the way forward if you want a modern game engine. 

We are keeping U3D's roadmap here: https://github.com/orgs/u3d-community/projects/1

Keeping the engine stable and working well across all supported platforms, and keeping third-party dependencies up-to-date and working on those platforms is already a lot of work. My personal focus right now is fostering a community where everyone is welcome and wants to keep the spirit Urho3D alive, but also drawing a clear line between the fork and the base (that is why the UI change was necessary).

This does not mean that U3D does not want new features: https://github.com/u3d-community/U3D/issues/1 is clearly one of them and we want more of those features, features that ultimately make game development easier. More IK, more built-in AI, more automation, more time-savers, without having to refactor the whole engine or big parts of it. 

As a community, we'll figure that roadmap out, discuss and iron out any issues, and slowly it will build it up to be U3D's shared vision. I invite everyone to create issues/tickets and give your opinion on existing ones, that would be awesome and a really valuable contribution.

**Reply by eugeneko** [reply]
> We are keeping U3D's roadmap here: https://github.com/orgs/u3d-community/projects/1

@klaussilveira btw this link is private or dead, I cannot access it.

**Reply by klaussilveira** [reply]
Oops. Fixed it.
